webapp/src/ts/modals/delete-doc-confirm/delete-doc-confirm.component.ts

```diff
@@ -1,40 +1,72 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
 import { BsModalRef } from 'ngx-bootstrap/modal';
 import * as LineageFactory from '@medic/lineage';
 import { TranslateService } from '@ngx-translate/core';
 import { Store } from '@ngrx/store';
+import { combineLatest, Subscription } from 'rxjs';
 
 import { DbService } from '@mm-services/db.service';
 import { MmModalAbstract } from '@mm-modals/mm-modal/mm-modal';
 import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
 
 @Component({
   selector: 'delete-doc-confirm',
   templateUrl: './delete-doc-confirm.component.html'
 })
-export class DeleteDocConfirmComponent extends MmModalAbstract implements OnInit {
-
+export class DeleteDocConfirmComponent extends MmModalAbstract implements OnInit, OnDestroy {
   private globalActions: GlobalActions;
+  subscriptions: Subscription = new Subscription();
+  selectMode;
   lineageLib;
   model = { doc: null }; // Automatically assigned by BsModalRef
 
   constructor(
     private store: Store,
     private translateService: TranslateService,
     private bsModalRef: BsModalRef,
-    private dbService: DbService
+    private dbService: DbService,
+    private router: Router
   ) {
     super();
     this.globalActions = new GlobalActions(this.store);
     this.lineageLib = LineageFactory(Promise, this.dbService.get());
   }
 
-  ngOnInit(): void { }
+  ngOnInit(): void {
+    const subscription = combineLatest(
+      this.store.select(Selectors.getSelectMode),
+    )
+    .subscribe(([
+      selectMode,
+    ]) => {
+      this.selectMode = selectMode;
+    });
+    this.subscriptions.add(subscription);
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
 
   close() {
     this.bsModalRef.hide();
   }
 
+  private getRouteName(url = '') {
+    const contacts = 'contacts';
+    const reports = 'reports';
+
+    if (url.includes(contacts)) {
```

Comment:
I think you might need to include `messages` here as well. 

If I load a conversation with one message and delete that message, I'm not redirected anywhere.

Reply:
The original code only redirect when it’s reports or contacts... if I add messages, then I need to listen to a selected conversation or pass over something from the global action that calls this modal and check if it's the last message before redirecting. I have added a [ToDo for improvement ](https://docs.google.com/spreadsheets/d/13BBw8cyV8CLMOzZAR_WDBqkaNrywNOk-i7b7pcypWJ8/edit#gid=303874733)

Reply:
Cool!